domain/modeldefaults/service/service.go

```diff
@@ -66,21 +76,135 @@ func (f ModelDefaultsProviderFunc) ModelDefaults(
 }
 
 // NewService returns a new Service for interacting with model defaults state.
-func NewService(st State) *Service {
+func NewService(
+	modelConfigProviderGetter ModelConfigProviderFunc,
+	st State,
+) *Service {
 	return &Service{
-		st: st,
+		modelConfigProviderGetter: modelConfigProviderGetter,
+		st:                        st,
+	}
+}
+
+// ProviderModelConfigGetter returns a [ModelConfigProviderFunc] for
+// retrieving provider based model config  values.
```

Comment:
Is this for ModelConfig or DefaultModelConfig?

Reply:
Both.